### DISCUSSION TITLE
Pendingcomponent is not displayed during routing.

### DISCUSSION THREAD

**Discussion by leesb971204** [TanStack/router#3058] [Q&A] [answered]
I believe that the pendingComponent should be displayed not only on a refresh but also during routing, as the data is being fetched and the routeState is in a pending state. 

https://stackblitz.com/edit/github-gh73by-7duz1xhf?file=src%2Froutes%2Fabout.tsx

When running the above example, I would expect the pendingComponent to be displayed when clicking on “About,” but it is not. This could lead users to think that the application has frozen. Is this the expected behavior?

**Comment by kaanozdokmeci** [comment]
in your stackblitz, it works when preload is disabled. but doesn't work with intent preload

**Reply by kaanozdokmeci** [reply, marked as answer]
this works fine on the latest version. 
the repro you have is on 1.45.7, current is 1.93.0

**Reply by leesb971204** [reply]
> this works fine on the latest version. the repro you have is on 1.45.7, current is 1.93.0

I have checked, thank you. Do you happen to know exactly which PR includes the changes? I searched for “pendingcomponent” on the release page, but nothing came up.

**Reply by kaanozdokmeci** [reply]
I don't, sorry.
